File: lite-app-starter/src/androidTest/java/com/chimbori/liteapp/starter/HermitIntentStarterTest.java
package com.chimbori.liteapp.starter;

import android.content.Intent;

import org.junit.Test;

import static com.chimbori.liteapp.starter.HermitIntentStarter.createSimpleLiteAppIntent;
import static org.junit.Assert.assertEquals;

public class HermitIntentStarterTest {
  @Test
  public void testCreateSimpleLiteAppIntent() {
    Intent actualIntent = createSimpleLiteAppIntent("https://example.org/");
    assertEquals(Intent.ACTION_VIEW, actualIntent.getAction());
    assertEquals("https://example.org/", actualIntent.getDataString());
    assertEquals("com.chimbori.hermitcrab/com.chimbori.hermitcrab.WebActivity", actualIntent.getComponent().flattenToString());
  }
}

File: lite-app-starter/src/androidTest/java/com/chimbori/liteapp/starter/HermitIntentStarterTest.kt
package com.chimbori.liteapp.starter

import android.content.Intent.ACTION_VIEW
import com.chimbori.liteapp.starter.HermitIntentStarter.createLiteAppIntent
import org.junit.Assert.assertEquals
import org.junit.Test

class HermitIntentStarterTest {
  @Test
  fun testCreateSimpleLiteAppIntent() {
    val actualIntent = createLiteAppIntent("https://example.org/")
    assertEquals(ACTION_VIEW, actualIntent.action)
    assertEquals("https://example.org/", actualIntent.dataString)
    assertEquals("com.chimbori.hermitcrab/com.chimbori.hermitcrab.WebActivity", actualIntent.component!!.flattenToString())
  }
}